pkg/services/folder/service.go

```diff
@@ -33,7 +35,10 @@ type Service interface {
 //go:generate mockery --name FolderStore --structname FakeFolderStore --outpkg foldertest --output foldertest --filename folder_store_mock.go
 type FolderStore interface {
 	// GetFolderByTitle retrieves a folder by its title
-	GetFolderByTitle(ctx context.Context, orgID int64, title string) (*Folder, error)
+	// It expects a folderUID as last argument.
+	// If folderUID is empty then the folder will be fetched from the root level
+	// otherwise it will be fetched from the subfolder under the folder with the given UID.
+	GetFolderByTitle(ctx context.Context, orgID int64, title string, folderUID *string) (*Folder, error)
```

Comment:
nit: same here, I think it would be easier to read if `folderUID` was called `parentUID` instead.